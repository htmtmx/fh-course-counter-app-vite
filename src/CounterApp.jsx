import PropTypes from 'prop-types';
import { useState } from 'react';

export const CounterApp = ({ title = 'No se ha proporcionado un tÃ­tulo', value = 0 }) => {

  const [counter, setCounter] = useState(value);

  const handleClick = (accionType) => {
    switch (accionType) {
      case 1:
        setCounter((c) => c + 1);
      break;
      case 2:
        setCounter(value);
      break;
      case 3:
        setCounter((c) => c - 1);
      break;
      default:
        break;
    }
  }

  return (
    <>
      <h1>{ title }</h1>
      <h2 data-testid='valueCounter'>{ counter }</h2>
      <button
        onClick={()=>handleClick(1)}
      >
        +1
      </button>
      <button aria-label='btn-reset'
        onClick={()=>handleClick(2)}
      >
        Reset
      </button>
      <button 
        onClick={()=>handleClick(3)}
      >
        -1
      </button>
    </>
  )
}

CounterApp.propTypes = {
  title: PropTypes.string,
  value: PropTypes.number.isRequired,
}