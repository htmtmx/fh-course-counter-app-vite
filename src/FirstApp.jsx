export const FirstApp = () => {
  return (
    <h1>Cesar</h1>
  )
}
