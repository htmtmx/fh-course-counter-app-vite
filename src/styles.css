html, body{
  background-color: #21232A;
  color: white;
  font-family: Helvetica, Arial, sans-serif;
  font-size: 1.3rem;
  padding: 70px;
}

button{
  background-color: #363636;
  border-radius: 5px;
  border: none;
  color: white;
  cursor: pointer;
  font-size: 1.3rem;
  margin: 1.5rem 0.5rem;
  padding: 10px 20px;
}