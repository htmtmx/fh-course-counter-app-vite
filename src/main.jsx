import React from 'react';
import ReactDOM from 'react-dom/client';
import { FirstApp } from './FirstApp';
import { ThemeProvider, createTheme } from '@mui/material/styles';
import CssBaseline from '@mui/material/CssBaseline';
import './styles.css'

const darkTheme = createTheme({
  palette: {
    mode: 'dark',
  },
});

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <ThemeProvider theme={darkTheme}>
      <CssBaseline/>
      <FirstApp title='Usando PropTypes' subtitle='Lorem ipsum dolor sit amet consectetur, adipisicing elit. Itaque omnis eius neque. Veritatis, vitae voluptas! Magni ad, soluta dolor optio earum accusamus odio impedit nam, quidem similique nobis, obcaecati id!' />
      
    </ThemeProvider>
  </React.StrictMode>
);