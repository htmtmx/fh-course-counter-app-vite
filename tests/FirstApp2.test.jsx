import { render, screen } from '@testing-library/react';
import { FirstApp } from '../src/FirstApp';
import { getAllByText } from '@testing-library/react';

describe('Prueba FirstApp2', () => {
  const title = 'Presentación';
  const name = 'Cesar';
  const subtitle = 'Soy un subtitulo';
  test('Debe de hacer match con el snapsot', () => { 

    const { container } = render(<FirstApp name={name} title={title} />);
    expect(container).toMatchSnapshot();
  });

  test('Debe de mostrar el mensaje Presentacion', () => {
    render(<FirstApp title={title} name={name} />);
    // screen.debug();
    expect(screen.getByText(title)).toBeTruthy();
  });

  test('Debe mostrar el titulo en un h1', () => {
    render(<FirstApp title={title} name={name} />);
    //! Forma básica de como resolver el test mediante un testId
    // expect(screen.getByTestId('test-title').innerHTML).toBe('Presentación');

    //! Forma de acceder al elemento HTML h1 y verificar el contenido del mismo
    expect(screen.getByRole('heading', {level: 1}).innerHTML).toContain(title);
  });
  test('Debe mostrar el subtitulo enviado por props', () => {
    render(<FirstApp title={title} name={name} subtitle={subtitle} />);
    expect(screen.getAllByText(subtitle).length).toBe(2);
  });
});