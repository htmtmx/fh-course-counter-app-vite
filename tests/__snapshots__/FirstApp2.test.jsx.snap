// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba FirstApp2 Debe de hacer match con el snapsot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Presentación
  </h1>
  <span>
    Cesar
  </span>
  <p>
    Sin subtítulo
  </p>
  <p>
    Sin subtítulo
  </p>
</div>
`;
